src/main/java/com/epam/aidial/core/controller/DeploymentPostController.java

```diff
@@ -292,9 +293,22 @@ void handleResponse() {
         context.setResponseBody(responseBody);
         context.setResponseBodyTimestamp(System.currentTimeMillis());
         Future<TokenUsage> tokenUsageFuture = Future.succeededFuture();
-        if (context.getDeployment() instanceof Model) {
+        if (context.getDeployment() instanceof Model model) {
             if (context.getResponse().getStatusCode() == HttpStatus.OK.getCode()) {
                 TokenUsage tokenUsage = TokenUsageParser.parse(responseBody);
+                if (tokenUsage == null) {
+                    Pricing pricing = model.getPricing();
```

Comment:
Why not always warn about missing token usage?

Reply:
there might be a model where cost is not calculated by token, e.g. by content lengths of response and requests. So the token usage is always null and we shouldn't log warning message in this case.